packages/fury-adapter-oas3-parser/lib/parser/oas/parseSchemaObject.js

```diff
@@ -58,10 +58,24 @@ function parseSchema(context) {
   const parseSubSchema = element => parseReference('schemas', R.uncurryN(2, parseSchema), context, element, true);
   const parseProperties = parseObject(context, `${name}' 'properties`, R.compose(parseSubSchema, getValue));
 
+  const arrayElementToParseResult = array => new namespace.elements.ParseResult(array.content);
+  const parseArray = itemParser => pipeParseResult(namespace,
+    R.unless(isArray, createWarning(namespace, `'${name}' 'required' is not an array`)),
+    R.compose(
```

Comment:
We're converting the array element into a parse result of elements from the array and then mapping those elements into a new parse result using the item parser. The item parser either returns an element or it returns an annotation to go into the result.

Then, we have to convert the successful values into an array element below.

Reply:
I wonder whether comments like these shouldn't be comments in the code :)